src/os_win.cc

```diff
@@ -326,11 +331,16 @@ char* OS::getenv(const char* variable) {
   wchar_t* variable_w = to_wide_string(variable);
 
   const int BUFFER_SIZE = 32767;
-  wchar_t buffer[BUFFER_SIZE];
+  wchar_t* buffer = unvoid_cast<wchar_t*>(malloc(BUFFER_SIZE));
   int length_w = GetEnvironmentVariableW(variable_w, buffer, BUFFER_SIZE);
   free(variable_w);
-  if (length_w == 0 || length_w > BUFFER_SIZE) return null;
-  return to_narrow_string(buffer, length_w);
+  if (length_w == 0 || length_w > BUFFER_SIZE) {
```

Comment:
This is different than the other check.

Reply:
See the other response.

I added a comment.